components/project-loader/project-loader.jsx

```diff
@@ -37,19 +35,17 @@ export default React.createClass({
     // we want to make sure exisiting fetch call is cancelled before we start a new one.
     // This prevents any unfinished request to incorrectly update the component state
     // when it returns.
-    if (this.state.tokenOfExistingPromise) {
-      this.state.tokenOfExistingPromise.cancel();
+    if (this.promiseToken) {
+      this.promiseToken.cancel();
     }
 
     let combinedParams = Object.assign({}, params, { page: this.state.apiPageIndex });
     let token = {};
 
-    this.setState({
-      // we are about to make a new request, set loadingData to true.
-      loadingData: true,
-      // we want component to know about the promise returned from the to-be-called Service.entries.get()
-      tokenOfExistingPromise: token
-    });
+    // we want component to know about the promise returned from the to-be-called Service.entries.get()
+    this.promiseToken = token;
```

Comment:
this is still a little strange: if we're binding this early, then we don't even need `token`, we can just set `this.promiseToken = {}` and then on line 51 say `.get(combinedParams, this.promiseToken)`. However, if that token needs to be "fresh" until the Service chain resolves, we don't actually want to say `this.promiseToken = token` here, yet.